### PR TITLE
Support podman based development environment

### PR DESCRIPTION
**Pull request by athos-ribeiro**
* Use host volumes
* Create shared directories with proper permissions
* Document the need to loosen SElinux restrictions

Signed-off-by: Athos Ribeiro <athos@redhat.com>

### CODE REVIEW

**Comment by athos-ribeiro** [comment]
I created this PR to show how I am running integration tests with podman-compose instead of docker.

If we were to merge this, we would need to move the entries in the shell script to the Makefile; losen specific selinux configurations instead of setting it to permissive mode; and possibly rebuild the Nexus image to avoid the need of changing permissions on the nexus-volume directory.

**Comment by athos-ribeiro** [comment]
Cc @lcarva, @nirzari, @chmeliik, @akhmanova, @ejegrova

File: prepare_podman_compose.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# make sure selinux is off
+sestatus | grep '^Current mode:.*enforcing$'
+if [ $? -eq 0 ]; then
+  echo "Selinux is enforcing.Set it to permissive mode before using podman-compose"
```
Comment (lcarva) [inline prepare_podman_compose.sh:6]: Yeah, I don't think podman-compose handles the `:z` modifier on volumes properly.

File: prepare_podman_compose.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  exit 1
+fi
+
+mkdir -p ./tmp/cachito-archives
```
Comment (lcarva) [inline prepare_podman_compose.sh:10]: Apparently this is an explicit backwards compatibility breakage, see https://github.com/containers/podman-compose/issues/185

It looks like a compatibility flag could be introduced to handle this.

Comment (athos-ribeiro) [reply prepare_podman_compose.sh:10]: Will this podman-compose patch also eliminate the need to change the nexus-volume permissions below?

Comment (lcarva) [reply prepare_podman_compose.sh:10]: Hmm unclear, but I was able to share named volumes between two containers in iib.

**Comment by athos-ribeiro** [comment]
- Rebased on master
- Moved setup script to Makefile

File: Makefile [hunk 3]
```diff
@@ -1,18 +1,35 @@
+export COMPOSE_ENGINE=docker-compose
```
Comment (lcarva) [inline Makefile:1]: Can we call this something like `CACHITO_COMPOSE_ENGINE` so I can set it in my `.bashrc` and forget about it? Then in this file we can set a default value if needed.

File: README.md [hunk 4]
```diff
@@ -77,6 +77,15 @@ Run the application locally (requires [docker-compose](https://docs.docker.com/c
 make run
 ```
 
+Alternatively, you could also run the application with
+[podman-compose](https://github.com/containers/podman-compose) by setting the
+`COMPOSE_ENGINE` variable (for now SELinux must be set to the **Permissive**
```
Comment (lcarva) [inline README.md:82]: Can we add a warning to indicate this is dangerous?

File: Makefile (hunk 3 above)
Comment (athos-ribeiro) [reply Makefile]: Done

File: README.md (hunk 4 above)
Comment (athos-ribeiro) [reply README.md]: Done. Also added it to the Makefile

**Comment by athos-ribeiro** [comment]
@lcarva, do you think this is good to be merged or should I close it and link this PR in the README file for whoever wants to run podman?

**Comment by athos-ribeiro** [comment]
Rebased on master

File: Makefile [hunk 5]
```diff
@@ -1,18 +1,35 @@
+CACHITO_COMPOSE_ENGINE ?= docker-compose
+ifeq ($(CACHITO_COMPOSE_ENGINE), docker-compose)
+	DOWN_OPTS=-v
```
Comment (lcarva) [inline Makefile:3]: If we're not using named volumes, you probably don't need `-v`, right? But it doesn't hurt.

**Review by lcarva — APPROVED** [review]
I think this is fine to be merged. The `podman-compose` approach is clearly documented.

**Pull request merged**